test/nuts/deployDestructive.nut.ts

```diff
@@ -0,0 +1,140 @@
+/*
+ * Copyright (c) 2020, salesforce.com, inc.
+ * All rights reserved.
+ * Licensed under the BSD 3-Clause license.
+ * For full license text, see LICENSE.txt file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+
+import * as path from 'path';
+import * as os from 'os';
+import { expect } from 'chai';
+import { execCmd } from '@salesforce/cli-plugins-testkit';
+import { SourceTestkit } from '@salesforce/source-testkit';
+import { exec } from 'shelljs';
+
+describe('source:delete NUTs', () => {
+  const executable = path.join(process.cwd(), 'bin', 'run');
+  let testkit: SourceTestkit;
+
+  const createApexClass = (apexName = 'myApexClass') => {
+    // create and deploy an ApexClass that can be deleted without dependency issues
+    const output = path.join('force-app', 'main', 'default', 'classes');
+    const pathToClass = path.join(testkit.projectDir, output, `${apexName}.cls`);
+    execCmd(`force:apex:class:create --classname ${apexName} --outputdir ${output}`);
+    execCmd(`force:source:deploy -m ApexClass:${apexName}`);
+    return { apexName, output, pathToClass };
+  };
+
+  const createManifest = (metadata: string, manifesttype: string) => {
+    execCmd(`force:source:manifest:create --metadata ${metadata} --manifesttype ${manifesttype}`);
+  };
+
+  const query = (
+    memberType: string,
+    memberName: string
+  ): { result: { records: Array<{ IsNameObsolete: boolean }> } } => {
+    return JSON.parse(
+      exec(
+        `sfdx force:data:soql:query -q "SELECT IsNameObsolete FROM SourceMember WHERE MemberType='${memberType}' AND MemberName='${memberName}'" -t --json`,
```

Comment:
one alternative is to create a Connection (since you have the username in testkit.project.setup from the org:create command) and use the query or singleRecordQuery features there.  It's *probably* faster than the shell route and you can use its <type> support